### ISSUE TITLE
Recursiveness wiki page violate SC2044

### ISSUE THREAD

**Issue by matsev** [koalaman/shellcheck#1521]
Problem:
The tips in the [Recursiveness](https://github.com/koalaman/shellcheck/wiki/Recursiveness) wiki page how to handle recursive usage of the `shellcheck` command violates the [SC2044](https://github.com/koalaman/shellcheck/wiki/SC2044) rule.

Proposed solution:
Rewrite the  [Recursiveness](https://github.com/koalaman/shellcheck/wiki/Recursiveness) page so that it obeys  [SC2044](https://github.com/koalaman/shellcheck/wiki/SC2044).

**Comment by nhooyr** [comment]
This is what I use:

```bash
mapfile -t scripts <<< "$(find . -type f -name "*.sh")"
shellcheck "${scripts[@]}"
```

**Comment by nhooyr** [comment]
Another option

```
bash -c 'shopt -s globstar; shellcheck **/*.{sh,ksh,bash}'
```

**Comment by nhooyr** [comment]
Actually, that doesn't work for me. From https://github.com/koalaman/shellcheck/wiki/TravisCI

**Comment by nhooyr** [comment]
```
nhooyr@ezekiel:*api-docs:~/Programming/opensource/ws
$ bash -c 'shopt -s globstar; shellcheck **/*.{sh,ksh,bash}'
**/*.sh: **/*.sh: openBinaryFile: does not exist (No such file or directory)
**/*.ksh: **/*.ksh: openBinaryFile: does not exist (No such file or directory)
**/*.bash: **/*.bash: openBinaryFile: does not exist (No such file or directory)
nhooyr@ezekiel:*api-docs:~/Programming/opensource/ws
$ bash -c 'shopt -s globstar; echo **/*.{sh,ksh,bash}'
**/*.sh **/*.ksh **/*.bash
```

Weird.

**Comment by matsev** [comment]
Please note that `mapfile` is not natively supported on OS X:

```
~$ mapfile
bash: mapfile: command not found
```

**Comment by nhooyr** [comment]
That's because macOS bash is super super old. I don't think we should worry about that. Just use home-brew and upgrade.

**Comment by koalaman** [comment]
You're right of course. I've updated them with POSIX compatible versions that check without warnings

**Comment by nhooyr** [comment]
The updated `shopt` example doesn't work for me still. The addition of `nullglob` just means every file gets sent to spellcheck when nothing matches.

**Comment by nhooyr** [comment]
Ah nvm, all is good. I think we do need to remove `nullglob` though.

**Comment by nhooyr** [comment]
Actually we do need it to prevent `**/*.` from showing up in the args.

**Comment by koalaman** [comment]
So what's the conclusion?

**Comment by nhooyr** [comment]
See the wiki page, I update it explaining all the options that are necessary.
